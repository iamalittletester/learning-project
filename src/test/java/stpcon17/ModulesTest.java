package stpcon17;

import browser.BrowserGetter;
import org.approvaltests.Approvals;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.PageFactory;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;
import pageobjects.stpcon17.ModulesPage;

import java.io.File;

public class ModulesTest {
    //webDriver instance
    private WebDriver webDriver;
    //class used for initializing browser
    private BrowserGetter browserGetter = new BrowserGetter();
    //page object class
    private ModulesPage page;

    @BeforeClass
    public void beforeClass() {
        //initialize the Chrome browser here
        webDriver = browserGetter.getChromeDriver();
        //initialize page object class
        page = PageFactory.initElements(webDriver, ModulesPage.class);
        webDriver.get(new File("src/test/resources/htmls/stpcon17/fullPage.html").getAbsolutePath());
    }

    @AfterClass
    public void afterClass() {
        webDriver.quit();
    }

    @Test
    public void images() {
        Approvals.verifyAll("",
                new Image[]{
                        new Image(page.readingModuleImageElement),
                        new Image(page.travelModuleImageElement),
                        new Image(page.photoModuleImageElement),
                }
        );
    }

    @Test
    public void links() {
        Approvals.verifyAll("",
                new Link[]{
                        new Link(page.readingModuleLinkElement),
                        new Link(page.travelModuleLinkElement),
                        new Link(page.photoModuleLinkElement),
                }
        );
    }

    @Test
    public void modules() {
        Approvals.verifyAll("",
                new Module[]{
                        new Module(page.readingModuleElement),
                        new Module(page.travelModuleElement),
                        new Module(page.photoModuleElement),
                }
        );
    }
}
