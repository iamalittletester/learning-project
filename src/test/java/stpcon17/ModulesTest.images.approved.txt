[0] = Image{src='https://c1.staticflickr.com/8/7494/27510788280_a9f43c3845_s.jpg', width='125', height='125'}
[1] = Image{src='https://farm2.staticflickr.com/1642/25881055090_e29679a37d_q.jpg', width='125', height='125'}
[2] = Image{src='https://farm1.staticflickr.com/612/32840145345_4bdf828d8f_q.jpg', width='125', height='125'}
